src/Microsoft.Azure.SignalR.Management/ServiceManagerBase.cs

```diff
@@ -57,53 +42,44 @@ public async Task<IServiceHubContext> CreateHubContextAsync(string hubName, ILog
                 case ServiceTransportType.Persistent:
```

Comment:
Is it possible to make the concrete ServiceManager as SingleTransientServiceManager / MultiTransientServiceManager  / SinglePersistentServiceManager / MultiPersistentServiceManager to remove the switch case?

Reply:
I think it's possible, but one Transport mode shares the client endpoint and access token generation logic ,and to reuse the code, we have to introduce SingleServiceManagerBase and MultiServiceManagerBase. I think the hierarchical structure is a little bit complicated.

Reply:
what about 
```
       
        interface ICreateContext
        {
            void CreateContext();
        }

        interface IGetClientInfo
        {
            void GetClientInfo();
        }

        interface IServiceManager: ICreateContext, IGetClientInfo
        {
        }

        // add other, shared part can be put into a new base / utility class
        class STG : IGetClientInfo
        {
            public void GetClientInfo()
            {
            }
        }

        // add other
        class STC : ICreateContext
        {
            public void CreateContext()
            {
            }
        }

        // todo: add other
        class ServiceManager : IServiceManager
        {
            ICreateContext _cc;
            IGetClientInfo _gci;
            public ServiceManager(ICreateContext cc, IGetClientInfo gci) 
            {
                _cc = cc;
                _gci = gci;
            }

            public void CreateContext()
            {
                _cc.CreateContext();
            }

            public void GetClientInfo()
            {
                _gci.GetClientInfo();
            }
        }


```